### PR TITLE
chore!: bump min node required to v14

### LINKED ISSUE #611: Gatsby build issue - 'regeneratorRuntime' is not defined
The project was working fine before using Content-management API. It's working fine over `gatsby develop` but the builds fails. The error refers to 'regeneratorRuntime' being undefined and I tried couple of it's solutions posted in the community it didn't solved my issue. Do share your views if you have faced anything like this or any thing that could help me out

Error:
```
success Building production JavaScript and CSS bundles - 35.999s
failed Building static HTML for pages - 10.501s
error Building static HTML failed

  35 | };
  36 | export var update = /*#__PURE__*/function () {
> 37 |   var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(http, params, rawData, headers) {
     | ^
  38 |     var _rawData$sys$version;
  39 | 
  40 |     var data;


  WebpackError: ReferenceError: regeneratorRuntime is not defined

  - app-definition.js:37 
    node_modules/contentful-management/dist/es-modules/plain/endpoints/app-definition.js:37:1

  - app-definition.js:64 
    node_modules/contentful-management/dist/es-modules/plain/endpoints/app-definition.js:64:2

  - index.js:1 
    node_modules/contentful-management/dist/es-modules/plain/endpoints/index.js:1:1

  - api-key.js:1 
    node_modules/contentful-management/dist/es-modules/entities/api-key.js:1:1

  - index.js:1 
    node_modules/contentful-management/dist/es-modules/entities/index.js:1:1

  - create-contentful-api.js:1 
    node_modules/contentful-management/dist/es-modules/create-contentful-api.js:1:1

  - contentful-management.js:1 
    node_modules/contentful-management/dist/es-modules/contentful-management.js:1:1

  - index.jsx:1 
    src/containers/suggestArticle/articleForm/index.jsx:1:1

  - index.jsx:1 
    src/containers/suggestArticle/index.jsx:1:1
```

Dependencies:
```
"dependencies": {
    "@contentful/rich-text-react-renderer": "^14.1.2",
    "@contentful/rich-text-types": "^14.1.2",
    "contentful-management": "^7.5.0",
    "formik": "^2.2.6",
    "gatsby": "^2.25.3",
    "gatsby-background-image": "^1.3.1",
    "gatsby-image": "^2.4.21",
    "gatsby-plugin-manifest": "^2.5.2",
    "gatsby-plugin-offline": "^3.3.3",
    "gatsby-plugin-postcss": "^3.1.0",
    "gatsby-plugin-react-helmet": "^3.3.14",
    "gatsby-plugin-sass": "^2.5.1",
    "gatsby-plugin-sharp": "^2.7.1",
    "gatsby-source-contentful": "^4.2.1",
    "gatsby-source-filesystem": "^2.4.2",
    "gatsby-transformer-sharp": "^2.5.21",
    "netlify-cli": "^2.69.9",
    "node-sass": "^4.14.1",
    "open-graph-scraper": "^4.7.1",
    "prop-types": "^15.7.2",
    "react": "^16.12.0",
    "react-dom": "^16.12.0",
    "react-helmet": "^6.1.0",
    "react-id-swiper": "^4.0.0",
    "react-input-range": "^1.3.0",
    "swiper": "^6.3.5",
    "yup": "^0.32.8"
  },
  "devDependencies": {
    "dotenv": "^8.2.0",
    "env-cmd": "^10.1.0",
    "prettier": "2.1.2",
    "tailwindcss": "^1.9.6"
  },
```

### PR DESCRIPTION
**Pull request by Spring3**
Resolves #611 

## Summary

Will be a breaking change hence is marked as a major release

* discontinue builds of legacy bundles to move on from IE
* bump the min node version to 16 which is lts at the time of writing
* remove `@deprecated` `maxDepth` param for references api as it was substituted with `include` to match the actual query param sent to the server

## Motivation and Context

Even microsoft moved away from supporting IE, so I think it is time to sunset the legacy bundle. I don't think someone is testing that it works here still either.

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [ ] There are no breaking changes
- [x] Changes are reflected in the documentation

When adding a new method:

- [x] The new method is exported through the default and plain CMA client
- [x] All new public types are exported from `./lib/export-types.ts`
- [x] Added a unit test for the new method
- [x] Added an integration test for the new method
- [x] The new method is added to the documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "types": "./dist/typings/contentful-management.d.ts",
   "module": "./dist/es-modules/contentful-management.js",
   "engines": {
-    "node": ">=6"
+    "node": ">=16"
```
Comment (catdalfonso) [inline package.json:10]: Node 14 isn't maintenance EOL until [2023-04-30](https://nodejs.org/en/about/releases/), and moreover is the latest supported node version on AWS Lambda - [this comment on the AWS repo suggests that v16 might be coming May 2022](https://github.com/aws/aws-lambda-base-images/issues/14#issuecomment-1100211561), but until then, any lambda-based apps will be unable to use this.

(Certainly think we need to move past node 6, though, and probably past 12 as well , which is maintenance EOL the end of this month.)

Comment (Spring3) [reply package.json]: Valid point. Using v14 then. Thanks

Comment (andipaetzold) [reply package.json]: Should we run the tests with node 14 as well?

Comment (Spring3) [reply package.json]: It was already at v16 before my PR so I can downgrade it but I don't know if version 16 was used intentionally or not.

Comment (Spring3) [reply package.json]: Using v14. As discussed, circleci orb will use v14 thanks [to its node/install command ](https://circleci.com/developer/orbs/orb/circleci/node#usage-install_nodejs)

Comment (Spring3) [reply package.json]: Screenshot 2022-04-20 at 13 58 04

Comment (andipaetzold) [reply package.json]: 👍 

We should now be able to use `cimg/base:stable` and downgrade the lockfile

Comment (Spring3) [reply package.json]: Fine. I kept the locked version just to avoid sudden bumps behind the "stable" tag. But make sense to use stable as well

**Review by shikaan — APPROVED** [review]

**Review by marcolink — APPROVED** [review]
🎉

**Pull request merged**